task-maker-async/src/client.rs

```diff
@@ -430,6 +433,233 @@ fn prepare_execution_group(
     Some(ret)
 }
 
+// TODO(veluca): consider using the tokio versions of file io.
+async fn read_file_to_memory(
+    store: &StoreClient,
+    file: &FileHandle,
+    size_limit: Option<usize>,
+) -> Result<Vec<u8>, Error> {
+    let mut result = vec![];
+    loop {
+        if let Some(limit) = size_limit {
+            if result.len() >= limit {
+                break;
+            }
+        }
+        let chunk = store
+            .read_chunk(context::current(), *file, result.len())
+            .await??;
+        match chunk {
+            FileReadingOutcome::Dropped => {
+                result.clear();
```

Comment:
When the file is dropped, the loop should exit, doesn't it?

Reply:
No, it shouldn't - when a file is dropped it will be recreated eventually.

Kinda a moot point since Dropped can't happen yet, but oh well.

Reply:
I see, fair enough.